src/SchemaGenerator.ts

```diff
@@ -151,6 +151,9 @@ export class SchemaGenerator {
                     return;
                 }
                 return;
+            case ts.SyntaxKind.ArrowFunction:
+                allTypes.set(`Parameters<typeof ${this.getFullName(node, typeChecker)}>`, node);
```

Comment:
I don't understand this part. Why do we add this specific type here?

Reply:
Otherwise it won't be able to find the `Parameters<typeof myFunc>` selector. If I run it without it the added test case will fail with `No root type "Parameters<typeof myFunction>" found`

Reply:
That sounds fishy. I don't have time to dig into it. Please investigate a bit to see what's going on.

Reply:
It makes sense to me, the current capabilities of the lib is to find "type names" of InterfaceDeclaration, ClassDeclaration, EnumDeclaration & TypeAliasDeclaration (which the prior switch cases handle). If we're introducing something akin to a type utility (`Parameters<typeof func>`) we need to introduce that as a "name". The other option I guess would be to just use the function name instead (`func`), although I'm not super fond of it since the schema entry defines the parameters and not the function per say. WDYT?

Reply:
Maybe we could just not support functions and only support params and return values

```ts
function foo(a: number, b: string) {
    return a;
}

// support
export type MyType = Parameters<typeof foo>;
export type MyType2 = ReturnType<typeof foo>;

// not support
export type MyType3 = typeof foo;
```

I haven't checked how easy that would be since we do need a function type node somewhere. WHat do you think?

Reply:
> Maybe we could just not support functions and only support params and return values

That would be possible, but let me try to explain my use case & why I want this feature in the lib. I am trying to use `ts-json-schema-generator` _programmatically_ to generate JSON schema input for ✨ some ✨ function (there's more to it, but this is how I plan to use your lib). If `ts-json-schema-generator` will only support type names then I have to impose the same restrictions on the users of my lib as well where all function parameters have to be type references. For the most simple functions it will get quite silly IMO:
```
type Num = number

function add(numberA: Num, numberB: Num) { // vs. having it inline `function add(numberA: number, numberB: number)`
  return numberA + numberB
}
```
I could, in theory, construct my own source file that adds arbitrary type names for the function parameters before parsing it with `ts-json-schema-generator`, but that feels quite hacky..

I hope you see why this would be a useful part of your lib and why it makes sense, if not I'd be happy to share more information